modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/serviceextensions/preupdatepassword/PreUpdatePasswordActionFailureTestCase.java

```diff
@@ -328,7 +339,112 @@ public void testUserSetPasswordViaOfflineInviteLink() throws Exception {
         scim2RestClient.deleteUser(offlineInvitingUserId);
     }
 
-    private void assertActionRequestPayload(String userId, String updatedPassword,
+    @Test(dependsOnMethods = "testUserSetPasswordViaOfflineInviteLink",
+            description = "Verify the admin initiated user registration with pre update password action failure")
+    public void testAdminInitiatedUserRegistration() throws Exception {
+
+        UserObject adminRegisteredUserInfo = new UserObject()
+                .userName(TEST_USER2_USERNAME)
+                .password(TEST_USER_PASSWORD)
+                .name(new Name().givenName(TEST_USER_GIVEN_NAME).familyName(TEST_USER_LASTNAME))
+                .addEmail(new Email().value(TEST_USER_EMAIL));
+        org.json.simple.JSONObject response = scim2RestClient.createUser(adminRegisteredUserInfo, null);
+        assertNotNull(response, "User creation response is null");
+
+        int statusCode = response.get("statusCode") != null
+                ? Integer.parseInt(response.get("statusCode").toString())
+                : -1;
+        String body = response.get("body") != null ? response.get("body").toString() : "";
+
+        org.json.simple.JSONObject bodyJson = null;
+        if (body != null && !body.isEmpty()) {
+            bodyJson = (org.json.simple.JSONObject) new org.json.simple.parser.JSONParser().parse(body);
+        }
+        int effectiveStatus = (bodyJson != null && bodyJson.get("status") != null)
+                ? Integer.parseInt(bodyJson.get("status").toString())
+                : statusCode;
+
+        assertTrue(
+                effectiveStatus == HttpServletResponse.SC_ACCEPTED
+                        || effectiveStatus == expectedPasswordUpdateResponse.getStatusCode(),
+                "Unexpected status. Raw response: " + response.toJSONString()
+        );
+
+        if (effectiveStatus == HttpServletResponse.SC_BAD_REQUEST) {
+            assertTrue(body.contains(expectedPasswordUpdateResponse.getErrorMessage()),
```

Comment:
Lets explicitly validate, the error message is contained in the `scimType` attribute of the error response.